Source/JavaScriptCore/b3/B3CheckSpecial.cpp

```diff
@@ -176,76 +176,90 @@ CCallHelpers::Jump CheckSpecial::generate(Inst& inst, CCallHelpers& jit, Generat
 
     context.latePaths.append(
         createSharedTask<GenerationContext::LatePathFunction>(
-            [=, this] (CCallHelpers& jit, GenerationContext& context) {
+            [=, this] (auto& jit, GenerationContext& context) {
                 fail.link(&jit);
 
                 // If necessary, undo the operation.
                 switch (m_checkKind.opcode) {
                 case BranchAdd32:
-                    if ((m_numCheckArgs == 4 && args[1] == args[2] && args[2] == args[3])
-                        || (m_numCheckArgs == 3 && args[1] == args[2])) {
-                        // This is ugly, but that's fine - we won't have to do this very often.
-                        ASSERT(args[1].isGPR());
-                        GPRReg valueGPR = args[1].gpr();
-                        GPRReg scratchGPR = CCallHelpers::selectScratchGPR(valueGPR);
-                        jit.pushToSave(scratchGPR);
-                        jit.setCarry(scratchGPR);
-                        jit.lshift32(CCallHelpers::TrustedImm32(31), scratchGPR);
-                        jit.urshift32(CCallHelpers::TrustedImm32(1), valueGPR);
-                        jit.or32(scratchGPR, valueGPR);
-                        jit.popToRestore(scratchGPR);
-                        break;
-                    }
-                    if (m_numCheckArgs == 4) {
-                        if (args[1] == args[3])
-                            Inst(Sub32, nullptr, args[2], args[3]).generate(jit, context);
-                        else if (args[2] == args[3])
-                            Inst(Sub32, nullptr, args[1], args[3]).generate(jit, context);
-                    } else if (m_numCheckArgs == 3)
-                        Inst(Sub32, nullptr, args[1], args[2]).generate(jit, context);
+                    // this instruction happens not to be used (and requires unimplemented assembler instructions) in 32-bit
+                    if constexpr (is64Bit()) {
```

Comment:
If constexpr doesn't disable compilation, so presumably this code would compile if you inverted this condition? Can we do if constexpr(!is64Bit()) UNREACHABLE_FOR_PLATFORM() instead?

Reply:
Or ASSERT(is64bit())? IS this code even incorrect? Can we just leave it in?